src/stores/A_store.js

```diff
@@ -36,8 +36,11 @@ class A_store extends ReduceStore {
                 const {x, y} = action;
                 const pathGrid = state.pathGrid.clone();
                 const cell = pathGrid.getCell(x, y);
-
-                cell.passable = !cell.passable;
+                if (cell.terrain === Cell.Terrain.WALL) {
```

Comment:
out of scope: maybe we should make this cycle through the types now that we have mountains 🤔